src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -131,15 +125,15 @@ internal class PicklingSqlCommandExecutor : SqlCommandExecutor
                     //   |null|
                     //   |  11|
                     //   +----+
-                    object[] inputRows = PythonSerDe.GetUnpickledObjects(readStream);
+                    object[] inputRows = PythonSerDe.GetUnpickledObjects(inputStream, messageLength);
 
                     for (int i = 0; i < inputRows.Length; ++i)
                     {
                         // Split id is not used for SQL UDFs, so 0 is passed.
                         outputRows.Add(commandRunner.Run(0, inputRows[i]));
                     }
 
-                    WriteOutput(outputStream, outputRows);
+                    WriteOutput(outputStream, outputRows, estimatedMaxSize: messageLength); // estimatedMaxSize is equal to messageLength because the produced output should be always smaller or of the same size as input
```

Comment:
Same here on the max chars in a line.

Also, I didn't get "because the produced output should be always smaller or of the same size as input". How can we guarantee this if you have an UDF like:
```C#
var udf = Udf<string, string>(str => str + str);
```
Am I misunderstanding this comment?

Reply:
I have changed the parameter name to "sizeHint" and simplified the comment.

The goal is to guess the max size in order to allocate the array only once.
The pickler is going to resize the buffer if needed (it's passed to the method by reference)